src/textual/css/styles.py

```diff
@@ -295,6 +333,71 @@ def get_transition(self, key: str) -> Transition | None:
         else:
             return None
 
+    def get_box_model(
+        self, container_size: Size, parent_size: Size
+    ) -> tuple[Size, Spacing]:
+        """Resolve the box model for this Styles.
+
+        Args:
+            parent_size (Size): The size of the widget container.
+            terminal_size (Size): The size of the terminal.
+
+        Returns:
+            tuple[Size, Spacing]: A tuple with the size of the content area and margin.
+        """
+        has_rule = self.has_rule
+        width, height = container_size
+
+        if has_rule("width"):
+            width = self.width.resolve_dimension(container_size, parent_size)
+        else:
+            width = max(0, width - self.margin.width)
+
+        if self.min_width:
+            min_width = self.min_width.resolve_dimension(container_size, parent_size)
+            width = max(width, min_width)
+
+        if self.max_width:
+            max_width = self.max_width.resolve_dimension(container_size, parent_size)
+            width = min(width, max_width)
+
+        if has_rule("height"):
+            height = self.height.resolve_dimension(container_size, parent_size)
+        else:
+            height = max(0, height - self.margin.height)
+
+        if self.min_height:
+            min_height = self.min_height.resolve_dimension(container_size, parent_size)
+            height = max(height, min_height)
+
+        if self.max_height:
+            max_height = self.max_height.resolve_dimension(container_size, parent_size)
+            height = min(width, max_height)
+
+        # TODO: box sizing
+
+        size = Size(width, height)
+        margin = Spacing(0, 0, 0, 0)
+
+        if self.box_sizing == "content-box":
+
+            if has_rule("padding"):
+                size += self.padding
+            if has_rule("border"):
+                size += self.border.spacing.totals
+            if has_rule("margin"):
+                margin = self.margin
+
+        else:  # border-box
+            if has_rule("padding"):
+                size -= self.padding
```

Comment:
Similar to the above comment, I think my expectation here would be that we would include the padding and border in the overall size rather than remove it, but I could be misunderstanding this code.

Reply:
Yeah, the size being calculated is the size that the Widget will render. The widget itself is responsible for rendering the border and padding.